plugins/modules/fabric_devices_workflow_manager.py

```diff
@@ -0,0 +1,4080 @@
+                        self.msg = (
+                            "The 'external_connectivity_ip_pool_name' is not a valid reserved pool for the "
+                            "device with IP '{ip}'.".format(ip=device_ip)
+                        )
+                        self.status = "failed"
+                        return self.check_return_status()
+
+                    l3_ip_handoff.update({
+                        "externalConnectivityIpPoolName": external_connectivity_ip_pool_name
+                    })
+                else:
+                    local_ip_address = item.get("local_ip_address")
+                    remote_ip_address = item.get("remote_ip_address")
+                    local_ipv6_address = item.get("local_ipv6_address")
+                    remote_ipv6_address = item.get("remote_ipv6_address")
+                    if not (local_ip_address and remote_ip_address):
+                        self.msg = (
+                            "The parameters 'local_ip_address' and 'remote_ip_address' is mandatory for "
+                            "creating L3 Handoff with IP Transit."
+                        )
+                        self.status = "failed"
+                        return self.check_return_status()
+
+                    l3_ip_handoff.update({
+                        "localIpAddress": local_ip_address,
+                        "remoteIpAddress": remote_ip_address
+                    })
+                    if local_ipv6_address and remote_ipv6_address:
+                        l3_ip_handoff.update({
+                            "localIpv6Address": local_ipv6_address,
+                            "remoteIpv6Address": remote_ipv6_address
+                        })
+
+            ip_l3_handoff_info.append(l3_ip_handoff)
+
+        return ip_l3_handoff_info
+
+    def get_want_fabric_devices(self, fabric_devices):
+        """
+        Get all the SDA fabric devices along with the Border settings, L2 Handoff,
+        L3 SDA Handoff, L3 IP Handoff information from playbook.
+        Set the status and the msg before returning from the API
+        Check the return value of the API with check_return_status()
+
+        Parameters:
+            fabric_devices (dict): Playbook details containing fabric devices details along
+            with the Border Settings, L2 Handoff, L3 SDA Handoff, L3 IP Handoff information.
+        Returns:
+            self: The current object with updated desired fabric devices information.
+        Description:
+            Do all the validation which is done in the GUI and format the payload
+            in such a way that we can pass it to the API.
+            Call the respective functions for gathering the device details,
+            L2 Handoff, L3 IP Handoff, L3 SDA Handoff provided by the user.
+        """
+
+        fabric_devices_details = []
+        fabric_name = fabric_devices.get("fabric_name")
+        device_config = fabric_devices.get("device_config")
```

Comment:
self.log("Starting to gather fabric device details for fabric: {fabric}".format(fabric=fabric_name), "DEBUG")